### PR TITLE
refectory: replace Seed() function to New() and NewSource() function

### PR DESCRIPTION
**Pull request by yj-yoo**
### What this PR does

The rand.Seed() function has been deprecated, and the official document recommends the rand.New(rand.NewSource) function.

The math/rand package now automatically seeds the global random number generator (used by top-level functions like Float64 and Int) with a random value, and the top-level Seed function has been deprecated. Programs that need a reproducible sequence of random numbers should prefer to allocate their own random source, using rand.New(rand.NewSource(seed)). -  https://tip.golang.org/doc/go1.20 -

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-launcher/virtwrap/converter/vcpu/vcpu_test.go [hunk 1]
```diff
@@ -24,7 +24,7 @@ var _ = Describe("VCPU pinning", func() {
 
 	BeforeEach(func() {
 		seed := time.Now().UnixNano()
-		rand.Seed(seed)
+		rand.New(rand.NewSource(seed))
```
[the remark this answers is not in the document]
Comment (yj-yoo) [reply pkg/virt-launcher/virtwrap/converter/vcpu/vcpu_test.go]: ok. I removed rand.New()

**Comment by yj-yoo** [comment]
> Thank you! The patch looks good. Can I ask you if you can also change the `rand.Seed(` occurrence here? Thank you
> 
> https://github.com/kubevirt/kubevirt/blob/3499434fa0a35b2f9dbcd1e924b10ddba95ea93b/tests/testsuite/fixture.go#L106


oh! You found something I missed! I would be very grateful if you could help us work together.

**Comment by yj-yoo** [comment]
> > > Thank you! The patch looks good. Can I ask you if you can also change the `rand.Seed(` occurrence here? Thank you
> > > https://github.com/kubevirt/kubevirt/blob/3499434fa0a35b2f9dbcd1e924b10ddba95ea93b/tests/testsuite/fixture.go#L106
> > 
> > 
> > oh! You found something I missed! I would be very grateful if you could help us work together.
> 
> Sure! I am checking the occurrences of `rand.Seed(` and there are just two of them. One that you already fix, and the other I mentioned that is still uncover, maybe just a leftover. Thank you!

Hi.
I found two rand.Seed()s in this issue.

One is "math/rand" that I modified, and I worked on it because Seed() of this package was deprecated.

Another one is the Seed() you mentioned, which is a function belonging to "k8s.io/apimachinery/pkg/util/rand" and is already processed internally as New(NewSource()), so no work is needed. see.

**Review by xpivarc — COMMENTED** [review]
/approve

**Pull request merged**